src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapper.tsx

```diff
@@ -0,0 +1,392 @@
+                            className="btn btn-default btn-xs"
+                            style={{ marginLeft: 5 }}
+                        >
+                            Select genes
+                        </button>
+                    </DefaultTooltip>
+                </div>
+                <div className="pathwayMapper">
+                    <div
+                        data-test="pathwayMapperTabDiv"
+                        className="cBioMode"
+                        style={{ width: '99%' }}
+                    >
+                        <Row>
+                            {/*
+                              // @ts-ignore */}
+                            <this.PathwayMapperComponent
+                                isCBioPortal={true}
+                                isCollaborative={false}
+                                genes={
+                                    this.props.store
+                                        .genesOfInterestForPathwayMapper as any
+                                }
+                                // newGenes = { this.newlyAddedGenes as any}
+                                cBioAlterationData={[]}
+                                genomicData={this.props.genomicData as any}
+                                tableComponent={this.renderTable}
+                                patientView={false}
+                                groupComparisonView={true}
+                                activeGroups={this.props.activeGroups as any}
+                                //  genesSelectionComponent = {this.renderGenesSelection}
+                            />
+                        </Row>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    changeGenesOfInterest(genes: string) {
+        let i: number = 0;
+        let newGeneOfInterest: AlterationEnrichmentRow[] = [];
+        let newGene: string = '';
+        let newGenes: string[] = [];
+        for (i = 0; i < genes.length; i++) {
+            if (
+                genes[i].charCodeAt(0) !== 10 &&
+                genes[i].charCodeAt(0) !== 32
+            ) {
+                newGene = newGene + genes[i].toUpperCase();
+                console.log(newGene);
+            } else {
+                console.log(newGene);
+                if (newGene !== '') newGenes.push(newGene);
+                newGene = '';
+            }
+        }
+        if (newGene !== '') newGenes.push(newGene);
+        console.log(newGenes);
+        newGeneOfInterest = this.props.genomicData.filter(gene => {
+            return this.doesInclude(gene.hugoGeneSymbol, newGenes);
+        });
+        console.log(newGeneOfInterest);
+        //this.genes = newGeneOfInterest;
+    }
+
+    doesInclude(gene: string, newGenes: string[]) {
+        let i: number = 0;
+        for (i = 0; i < newGenes.length; i++) {
+            if (newGenes[i] === gene) return true;
+        }
+        return false;
+    }
```

Comment:
Do not rewrite basic `find` functionality. You can use one of the `lodash` functions for this.